src/components/AuthContainer.ts

```diff
@@ -15,11 +15,11 @@ import TermsOfService from 'src/pages/TermsOfService';
 const AuthContainer = ({ children }) => {
   const { name, public: isPublic } = useRoute();
   const { isSignedIn, registrationStatus, termsOfService, profile } = useStore(authStore);
-  const displayTosPage = isSignedIn && termsOfService.permitsSystemUsage === false;
+  const displayTosPage = isSignedIn === true && (termsOfService as any).permitsSystemUsage === false;
   const seenAzurePreview = useStore(azurePreviewStore) || false;
-  const authspinner = () => h(centeredSpinner, { style: { position: 'fixed' } });
+  const authspinner = () => h(centeredSpinner, div({ style: { position: 'fixed' } }));
 
-  return Utils.cond(
+  return Utils.cond<any>(
```

Comment:
Wherever possible, let's avoid explicit `any`. It looks like `ReactNode` (imported from `react`) should work here.

```suggestion
  return Utils.cond<ReactNode>(
```